source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2449,16 +2449,33 @@ namespace TrilinosWrappers
 {
   namespace internal
   {
+    namespace
+    {
+#ifndef DEAL_II_WITH_MPI
```

Comment:
Thanks for the confirmation! Apologies that I didn't make the comment several lines further down. 

One thing that I'm curious about though: does `#ifndef DEAL_II_WITH_MPI` strictly mean that `Trilinos` is built without `MPI`? Can we not link to, for example, a serial build of `Trilinos` but a parallel build of `PETSc`?